### PR TITLE
Use greedy label overlap prevention for log scales.

### PR DESCRIPTION
**Pull request by domoritz**

### CODE REVIEW

**Comment by kanitw** [comment]
Code looks reasonable -- any screenshot?

**Comment by kanitw** [comment]
Also need to re-run build:examples :)

**Comment by domoritz** [comment]
I didn't want to make the merge harder for @arvind. This pr is for his branch.

**Comment by kanitw** [comment]
okie :)

File: src/compile/axis/rules.ts [hunk 1]
```diff
@@ -136,9 +137,11 @@ export function domainAndTicks(property: keyof VgAxis, specifiedAxis: Axis, isGr
   return specifiedAxis[property];
 }
 
-export function labelOverlap(fieldDef: FieldDef<string>, specifiedAxis: Axis, channel: Channel, isGridAxis: boolean) {
-  // TODO: use true for non-log continuous scales, and use "greedy" for log scales
+export function labelOverlap(fieldDef: FieldDef<string>, specifiedAxis: Axis, channel: Channel, isGridAxis: boolean, scaleType: ScaleType) {
   if (!isGridAxis && channel === 'x' && !labelAngle(specifiedAxis, channel, fieldDef)) {
+    if (scaleType === 'log') {
```
Comment (kanitw) [inline src/compile/axis/rules.ts]: @domoritz Actually why is this rule for log only applied for x?  (I guess y's log can similar has overlapping labels?)

**Pull request merged**